### ISSUE TITLE
rook-api should use prometheus annotations

### ISSUE THREAD

**Issue by jpds** [rook/rook#1091]
rook-api currently uses this[1] configuration to configure a monitoring endpoint. However it would be better if the service definition had these annotations from here in its metadata:

- https://github.com/prometheus/prometheus/blob/5ab8834befbd92241a88976c790ace7543edcd59/documentation/examples/prometheus-kubernetes.yml#L133

This way a Prometheus instance that's monitoring kubernetes-services can autodetect the rook-api service metrics.

[1]: https://github.com/rook/rook/blob/c6bb14ad28c5540e536e3bc8e838b9dda80f5409/cluster/examples/kubernetes/monitoring/service-monitor.yaml#L13

**Comment by jpds** [comment]
I found an easy way to enable this:

```
kubectl -n rook annotate svc rook-api prometheus.io/port=8124
kubectl -n rook annotate svc rook-api prometheus.io/scrape=true
```

However my Prometheus instance deployed through the `stable/prometheus` chart isn't able to poll the /metrics URL.

**Comment by jpds** [comment]
I've tried this with `curl` from the `rook-tools` container and it seems that prometheus is timing out on the endpoint before the values are successfully returned.

**Comment by jbw976** [comment]
hitting the `/metrics` endpoint on the API pod from the toolbox pod with `curl` is timing out @jpds?  Does `rookctl status` show any issues with the cluster?  Do the API pod logs show any calls taking a very long time to complete?

**Comment by jpds** [comment]
`curl` and `rookctl status` from the `rook-tools` on occasion fail talking to the `rook-api` endpoint:

```
$ kubectl exec -it -n rook rook-tools -- rookctl status
failed to get status: Get http://x.x.x.x:8124/status: net/http: request canceled (Client.Timeout exceeded while awaiting headers)
```

However if I try again a few seconds later, everything is fine and `status` shows that the cluster is fine.

The API logs sometimes show:

```
api: 61d73bdc-bb68-4202-bf09-441442e1ebdc	GET	/metrics	GetMetrics	200 OK	1m5.637952332s
...
api: a5be2b81-1402-4d02-9207-26e77b727295	GET	/status	GetStatusDetails	200 OK	16.66156607s
```

I think Prometheus times out at 10s...

**Comment by jpds** [comment]
This little trick appears to have helped:

```
kubectl -n rook annotate svc rook-api prometheus.io/scrape_timeout=2m
kubectl -n rook annotate svc rook-api prometheus.io/scrape_interval=5m
```

**Comment by jbw976** [comment]
wow, those response times are WAY longer than i would expect.  I wonder what is taking so long inside the API pod before it returns a response finally.  Do you have the full api pod log to share?

**Comment by jpds** [comment]
Checked over the log and saw that it was using an old mon map, filed #1102 about this. Now my `/metrics` endpoint returns in 2s.

**Comment by travisn** [comment]
@jpds should this be closed in favor of #1111?

**Comment by jpds** [comment]
@travisn Yes, indeed!
